### PR TITLE
optimize: add StateType Enum

### PR DESCRIPTION
**Pull request by MaoMaoandSnail**
- [ ] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did
optimize: computeIfAbsent get twice
optimize: createProcessorController code order
optimize: add StateType Enum

### CODE REVIEW

File: common/src/main/java/org/apache/seata/common/util/CollectionUtils.java [hunk 1]
```diff
@@ -271,11 +267,17 @@ public static Map<String, String> decodeMap(String data) {
...
     public static <K, V> V computeIfAbsent(Map<K, V> map, K key, Function<? super K, ? extends V> mappingFunction) {
-        V value = map.get(key);
-        if (value != null) {
-            return value;
+        V value;
+        if ((value = map.get(key)) == null) {
+            Objects.requireNonNull(mappingFunction);
+            V newValue;
+            if ((newValue = mappingFunction.apply(key)) != null) {
+                map.put(key, newValue);
```
Comment (funky-eyes) [inline common/src/main/java/org/apache/seata/common/util/CollectionUtils.java]: ConcurrentHashMap#computeIfAbsent ，被你这个改动后，岂不是线程不安全了？
After being changed by you, isn't the thread unsafe?

Comment (MaoMaoandSnail) [reply common/src/main/java/org/apache/seata/common/util/CollectionUtils.java]: you are right

File: common/src/main/java/org/apache/seata/common/util/CollectionUtils.java [hunk 2]
```diff
@@ -16,11 +16,7 @@
  */
 package org.apache.seata.common.util;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (funky-eyes) [inline common/src/main/java/org/apache/seata/common/util/CollectionUtils.java]:
```suggestion
import java.util.ArrayList;
import java.util.Collection;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
```

Comment (MaoMaoandSnail) [reply common/src/main/java/org/apache/seata/common/util/CollectionUtils.java]: OK

File: saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/StateMachineProcessRouter.java [hunk 3]
```diff
@@ -105,19 +106,21 @@ public Instruction route(ProcessContext context) throws FrameworkException {
...
-            this.stateRouters.put(DomainConstants.STATE_TYPE_SUB_MACHINE_COMPENSATION, taskStateRouter);
-            this.stateRouters.put(DomainConstants.STATE_TYPE_LOOP_START, taskStateRouter);
-
-            this.stateRouters.put(DomainConstants.STATE_TYPE_SUCCEED, new EndStateRouter());
-            this.stateRouters.put(DomainConstants.STATE_TYPE_FAIL, new EndStateRouter());
+        if (!stateRouters.isEmpty()) {
+            return;
         }
+
+        TaskStateRouter taskStateRouter = new TaskStateRouter();
```
Comment (slievrly) [inline saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/StateMachineProcessRouter.java:113]: So many getValues are not very elegant, it is recommended to centralize them in the method.

Comment (MaoMaoandSnail) [reply saga/seata-saga-engine/src/main/java/org/apache/seata/saga/engine/pcext/StateMachineProcessRouter.java:113]: OK

File: saga/seata-saga-spring/src/main/java/org/apache/seata/saga/engine/store/db/DbAndReportTcStateLogStore.java [hunk 4]
```diff
@@ -48,11 +48,7 @@
 import org.apache.seata.saga.engine.store.StateLogStore;
 import org.apache.seata.saga.engine.tm.SagaTransactionalTemplate;
 import org.apache.seata.saga.proctrl.ProcessContext;
-import org.apache.seata.saga.statelang.domain.DomainConstants;
-import org.apache.seata.saga.statelang.domain.ExecutionStatus;
-import org.apache.seata.saga.statelang.domain.StateInstance;
-import org.apache.seata.saga.statelang.domain.StateMachine;
-import org.apache.seata.saga.statelang.domain.StateMachineInstance;
+import org.apache.seata.saga.statelang.domain.*;
```
Comment (slievrly) [inline saga/seata-saga-spring/src/main/java/org/apache/seata/saga/engine/store/db/DbAndReportTcStateLogStore.java]: not allow import *

**Comment by slievrly** [comment]
statelang/src/main/java/org/apache/seata/saga/statelang/parser/utils/StateMachineUtils.java:21:8: Unused import - org.apache.seata.saga.statelang.domain.StateType. [UnusedImports]

File: saga/seata-saga-spring/src/main/java/org/apache/seata/saga/engine/store/db/DbAndReportTcStateLogStore.java (hunk 4 above)
Comment (MaoMaoandSnail) [reply saga/seata-saga-spring/src/main/java/org/apache/seata/saga/engine/store/db/DbAndReportTcStateLogStore.java]: I removed it

Comment (MaoMaoandSnail) [reply saga/seata-saga-spring/src/main/java/org/apache/seata/saga/engine/store/db/DbAndReportTcStateLogStore.java]: I rollback it

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**